### PR TITLE
Returns error on invalid request line

### PR DESCRIPTION
**Pull request by catenacyber**
Fixes #2655

Follows #190

### CODE REVIEW

File: test/files/42-http_0_9-method_only.t [hunk 1]
```diff
@@ -1,3 +1,2 @@
 >>>
-HELLO
```
Comment (victorjulien) [inline test/files/42-http_0_9-method_only.t:2]: why was this test replaced? It seems like a very different thing to test?

Comment (catenacyber) [reply test/files/42-http_0_9-method_only.t:2]: HTTP 0.9 has only one method and it is `GET`

Maybe we can rename the test instead of changing its content

Comment (victorjulien) [reply test/files/42-http_0_9-method_only.t:2]: Yeah lets leave the old test in place as well. It should be handled in some way as well right? Or was the handling wrong?

**Comment by victorjulien** [comment]
```
htp_request.c: In function ‘htp_connp_REQ_FINALIZE’:
htp_request.c:804:21: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         while ((pos < connp->in_current_len) && htp_is_space(connp->in_current_data[pos]))
                     ^
htp_request.c:806:17: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (pos < connp->in_current_len) {
                 ^
htp_request.c:810:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             while ((pos < connp->in_current_len) && (!htp_is_space(connp->in_current_data[pos])))
                         ^
```

File: test/files/42-http_0_9-method_only.t (hunk 1 above)
Comment (catenacyber) [reply test/files/42-http_0_9-method_only.t:2]: Ok, I am renaming the old test, and adding a new with real HTTP 0.9

**Comment by catenacyber** [comment]
Thanks for the warnings report

**Pull request closed without merging**